src/main/Config.cpp

```diff
@@ -264,6 +264,7 @@ Config::Config() : NODE_SEED(SecretKey::random())
     MAX_DEX_TX_OPERATIONS_IN_TX_SET = std::nullopt;
 
     ENABLE_SOROBAN_DIAGNOSTIC_EVENTS = false;
+    ENABLE_DIAGNOSTICS_FOR_TX_SUBMISSION = true;
```

Comment:
Should this be enabled by default? Do we need the tier 1 validators to have this enabled? I was under the impression that transactions usually aren't submitted directly to tier 1. In that case, it'd be safer if this defaulted to false.